meta-mbl-distro/scripts/lib/mbl/util/fileutil.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+"""Utility functions for handling files."""
+
+import inspect
+import pathlib
+
+
+def strict_path_resolve(path):
+    """
+    Resolve a path.
+
+    Acts like
+        pathlib.Path(path).resolve(strict=True)
+    from Python >= 3.6.
+    """
+    if "strict" in inspect.getfullargspec(path.resolve).args:
+        # In python >= 3.6, pathlib.Path.resolve() has a "strict"
+        # parameter, and we must set it to True to avoid the default
+        # value of False.
+        return pathlib.Path(path).resolve(strict=True)
+    # In python < 3.6, pathlib.Path.resolve() doesn't have a "strict"
+    # parameter, but it always does strict resolution.
+    return pathlib.Path(path).resolve()
+
+
+def read_chunks(f):
+    """Read 4KiB chunks from a file-like object."""
```

Comment:
Any specific reason why we are using 4kiB? Is this machine dependant? If so I suggest we add a parameter for the chunk size and defaults to `4096`, if not I suggest renaming the function do `read_4kiB_chunks`.

Reply:
I'll add a chunk size parameter. I don't really want the user to have to care what the chunk size is though - I'll add a comment explaining the choice of `4096` as a default.

Reply:
https://github.com/ARMmbed/meta-mbl/pull/763/commits/51e3be7f15a01865d93201b0db18755c87dbf282